### PR TITLE
Reduce Retry Time And Add imds Fallback Client

### PR DESCRIPTION
**Pull request by sethAmazon**
**Description:** 
Reduce imds retry time to match downstream. 

**Link to tracking Issue:** 
N/A

**Testing:** 
Tested downstream retry. This code is copied.

**Documentation:** 
N/A

### CODE REVIEW

**Review by okankoAMZ — DISMISSED** [review]

File: override/aws/imdsretryer.go [hunk 1]
```diff
@@ -65,5 +66,8 @@ func (r iMDSRetryer) ShouldRetry(req *request.Request) bool {
 	if awsError, ok := req.Error.(awserr.Error); r.DefaultRetryer.ShouldRetry(req) || (ok && awsError != nil && awsError.Code() == "EC2MetadataError") {
 		shouldRetry = true
 	}
+	if r.logger != nil {
+		r.logger.Debug("imds error : ", zap.Bool("shouldRetry", shouldRetry), zap.Error(req.Error))
```
Comment (okankoAMZ) [inline override/aws/imdsretryer.go:70]: why are we using zap?

Comment (sethAmazon) [reply override/aws/imdsretryer.go:70]: That is the logger otel contrib uses

**Review by okankoAMZ — APPROVED** [review]

File: exporter/datadogexporter/internal/hostmetadata/internal/ec2/ec2.go [hunk 2]
```diff
@@ -68,19 +68,37 @@ func GetHostInfo(logger *zap.Logger) (hostInfo *HostInfo) {
...
 		hostInfo.EC2Hostname = ec2Hostname
 	} else {
-		logger.Warn("Failed to get EC2 hostname", zap.Error(err))
+		childCtxFallbackEnable, cancelRetryEnable := context.WithTimeout(ctx, override.TimePerCall)
+		defer cancelRetryEnable()
+		if ec2HostnameInner, errInner := metaFallbackEnable.GetMetadataWithContext(childCtxFallbackEnable, "hostname"); errInner == nil {
+			hostInfo.EC2Hostname = ec2HostnameInner
+		} else {
+			logger.Warn("Failed to get EC2 hostname", zap.Error(err))
+		}
```
Comment (adam-mateen) [inline exporter/datadogexporter/internal/hostmetadata/internal/ec2/ec2.go:101]: Do we really need to change the datadog exporter?
Do we somehow rely on code from this exporter in ours?

Comment (sethAmazon) [reply exporter/datadogexporter/internal/hostmetadata/internal/ec2/ec2.go:101]: No we do not. I just did a search for where imds was called and added the code there. I can remove this if need be, but I think it is fine to keep it.

File: override/aws/imdsretryer.go [hunk 3]
```diff
@@ -33,28 +34,28 @@ import (
 var IMDSRetryer request.Retryer = newIMDSRetryer()
 
 const (
-	TimePerCall = 2 * time.Minute
+	TimePerCall = 30 * time.Second
```
Comment (adam-mateen) [inline override/aws/imdsretryer.go:37]: What is the context for reducing this TimePerCall?
Business decision?

Comment (sethAmazon) [reply override/aws/imdsretryer.go:37]: We had a flaky test for 2 minutes in the private. I changed it to 30 seconds there so I changed it to 30 seconds here to match.

**Review by adam-mateen — APPROVED** [review]

**Pull request merged**